### PR TITLE
2060  add notes about inserting nan inf and -inf to starrocks, doris, and clickhouse 

### LINKED ISSUE #2060: Document: fix(sink): fix starrocks doris and clickhouse decimal
This issue tracks the documentation update needed for the merged PR #15664.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/15664
Source PR Merged At: 2024-04-11T08:41:47Z

### PR DESCRIPTION
**Pull request by ShanlanLi**
## Info

- **Description**

  - [ What's changed? Which parts of the docs are affected? ]

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - https://github.com/risingwavelabs/risingwave/pull/15664

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/2060

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/guides/sink-to-doris.md [hunk 1]
```diff
@@ -102,4 +102,10 @@ In regards to `decimal` types, RisingWave will round to the nearest decimal plac
 |ARRAY | ARRAY |
 |No support | BYTEA |
 |JSONB | JSONB |
-|BIGINT | SERIAL |
\ No newline at end of file
+|BIGINT | SERIAL |
+
+:::note
+
+Previously, when inserting data into the Apache Doris sink, an error would be reported if the values were "nan (not a number)", "inf (infinity)", or "-inf (-infinity)". However, we have now made a change to the behavior. If a decimal value is out of bounds or represents "inf", "-inf", or "nan", we will insert null values. Please note this modification to how these special values are handled.
```
Comment (ShanlanLi) [inline docs/guides/sink-to-doris.md]: @xxhZs Just to make sure, for Doris, have we removed the previous logic for checking the maximum number of bits before insertion? If we have, I'll add notes about this logic change here.

File: docs/guides/sink-to-starrocks.md [hunk 2]
```diff
@@ -90,4 +90,16 @@ The following table shows the corresponding data type in RisingWave that should
...
+:::note
+
+Previously, when inserting data into the StarRocks sink, an error would be reported if the values were "nan (not a number)", "inf (infinity)", or "-inf (-infinity)". However, we have now made a change to the behavior. If a decimal value is out of bounds or represents "inf", "-inf", or "nan", we will insert null values. Please note this modification to how these special values are handled.
+
+In addition, with the recent updates in StarRocks, we have modified the calibration logic for decimal values. 
+
+Previously, we had to make precision adjustments to match StarRocks' specified precision. For example, if we had a value like `1.123456` and StarRocks had a precision of 3, we would convert it to `1.123`. However, the latest versions of StarRocks can now automatically handle this conversion, eliminating the need for manual adjustments. 
+
+we have also removed the previous logic for checking the maximum number of bits before insertion. There is no longer a need to perform a pre-insertion check on the number of bits and report errors. If a value exceeds the maximum number of supported bits, it cannot be inserted into the downstream system.
+
```
Comment (xxhZs) [inline docs/guides/sink-to-starrocks.md]: > In addition, with the recent updates in StarRocks, we have modified the calibration logic for decimal values. 
> 
> Previously, we had to make precision adjustments to match StarRocks' specified precision. For example, if we had a value like `1.123456` and StarRocks had a precision of 3, we would convert it to `1.123`. However, the latest versions of StarRocks can now automatically handle this conversion, eliminating the need for manual adjustments. 
> 
> we have also removed the previous logic for checking the maximum number of bits before insertion. There is no longer a need to perform a pre-insertion check on the number of bits and report errors. If a value exceeds the maximum number of supported bits, it cannot be inserted into the downstream system.

There is no need to specify here, this is just a change in our internal logic, For users, it is consistent with before

File: docs/guides/sink-to-clickhouse.md [hunk 3]
```diff
@@ -205,6 +205,12 @@ In ClickHouse, the `Nested` data type doesn't support multiple levels of nesting
 
 :::
 
+:::note
+
+Previously, when inserting data into the ClickHouse sink, an error would be reported if the values were "nan (not a number)", "inf (infinity)", or "-inf (-infinity)". However, we have now made a change to the behavior. If the ClickHouse column is nullable, we will insert null values in such cases. If the column is not nullable, we will insert `0` instead. Please note this modification to how these special values are handled.
```
Comment (emile-00) [inline docs/guides/sink-to-clickhouse.md]:
```suggestion
Previously, when inserting data into a ClickHouse sink, an error would be reported if the values were "nan (not a number)", "inf (infinity)", or "-inf (-infinity)". However, we have made a change to this behavior. If the ClickHouse column is nullable, we will insert null values in such cases. If the column is not nullable, we will insert `0` instead. 
```

File: docs/guides/sink-to-doris.md (hunk 1 above)
Comment (emile-00) [inline docs/guides/sink-to-doris.md]:
```suggestion
Previously, when inserting data into an Apache Doris sink, an error would be reported if the values were "nan (not a number)", "inf (infinity)", or "-inf (-infinity)". However, we have made a change to the behavior. If a decimal value is out of bounds or represents "inf", "-inf", or "nan", we will insert null values. 
```

File: docs/guides/sink-to-starrocks.md [hunk 4]
```diff
@@ -90,4 +90,16 @@ The following table shows the corresponding data type in RisingWave that should
 | ARRAY | ARRAY |
 | No support | BYTEA |
 | JSON | JSONB |
-| BIGINT | SERIAL |
\ No newline at end of file
+| BIGINT | SERIAL |
+
+:::note
+
+Previously, when inserting data into the StarRocks sink, an error would be reported if the values were "nan (not a number)", "inf (infinity)", or "-inf (-infinity)". However, we have now made a change to the behavior. If a decimal value is out of bounds or represents "inf", "-inf", or "nan", we will insert null values. Please note this modification to how these special values are handled.
```
Comment (emile-00) [inline docs/guides/sink-to-starrocks.md]:
```suggestion
Previously, when inserting data into a StarRocks sink, an error would be reported if the values were "nan (not a number)", "inf (infinity)", or "-inf (-infinity)". However, we have made a change to the behavior. If a decimal value is out of bounds or represents "inf", "-inf", or "nan", we will insert null values. 
```

File: docs/guides/sink-to-starrocks.md [hunk 5]
```diff
@@ -90,4 +90,16 @@ The following table shows the corresponding data type in RisingWave that should
...
+
+In addition, with the recent updates in StarRocks, we have modified the calibration logic for decimal values. 
+
+Previously, we had to make precision adjustments to match StarRocks' specified precision. For example, if we had a value like `1.123456` and StarRocks had a precision of 3, we would convert it to `1.123`. However, the latest versions of StarRocks can now automatically handle this conversion, eliminating the need for manual adjustments.
```
Comment (emile-00) [inline docs/guides/sink-to-starrocks.md]:
```suggestion
Previously, we had to make precision adjustments to match StarRocks' specified precision. For example, if we had a value like `1.123456` and StarRocks had a precision of 3, we would convert it to `1.123`. However, the latest versions of StarRocks now automatically handle this conversion, eliminating the need for manual adjustments. 
```

**Review by WanYixian — APPROVED** [review]
LGTM!

**Pull request merged**